doc/source/cluster/cluster_under_construction/ray-clusters-on-vms/user-guides/launching-clusters/on-premises.md

```diff
@@ -0,0 +1,116 @@
+# Launching On-Premises Cluster
+
+This docs describes how to set up an on-premises Ray cluster, i.e., to run Ray on bare metal machines, or in a private cloud. We provide two ways to start an on-premises cluster.
+
+* You can **manually set up** the Ray cluster by installing the Ray package and starting the Ray processes on each node. 
+* Alternatively, if you know all the nodes in advance and have ssh access to them, you should start the ray cluster using **cluster-launcher**.
+
+
+## Manually Set up a Ray Cluster 
+This section assumes that you have a list of machines and that the nodes in the cluster can communicate with each other. It also assumes that Ray is installed on each machine.  You can use pip to install it, or follow [install ray](https://docs.ray.io/en/latest/ray-overview/installation.html) for more detailed instructions.
+
+```
+# install ray cluster launcher, which is part of ray command line tool.
+pip install -U ray[default]
+```
+
+
+### Start Head Node
+On the head node (just choose one node to be the head node), run the following. If the --port argument is omitted, Ray will choose port 6379, falling back to a random port.
+
+```
+ray start --head --port=6379
+
+# To connect to this Ray runtime from another node, run
+ray start --address='<ip address>:6379'
+```
+
+The command will print out the address of the Ray GCS server that was started (the local node IP address plus the port number you specified). If connection fails, check your firewall settings and network configuration.
+
+### Start Worker Nodes
+Then on each of the other nodes, run the following command to connect to the head node you just created.
+
+```
+ray start --address=<head-node-address:port>
+
+```
+Make sure to replace `head-node-address:port` with the value printed by the command on the head node (it should look something like 123.45.67.89:6379).
+
+Note that if your compute nodes are on their own subnetwork with Network Address Translation, to connect from a regular machine outside that subnetwork, the command printed by the head node will not work. You need to find the address that will reach the head node from the second machine. If the head node has a domain address like compute04.berkeley.edu, you can simply use that in place of an IP address and rely on the DNS.
```

Comment:
`find an address` since there might be more than one. 

...the berkeley address has tipped me off to the fact that this content already exists and has been copied from somewhere else.

Going to stop editing the grammar for now.